tests/lib/eslint/eslint.js

```diff
@@ -994,11 +999,265 @@ describe("ESLint", () => {
                 assert.strictEqual(results[0].messages[0].ruleId, "no-undef");
             });
         });
+
+        describe("TypeScript config files", () => {
+
+            const flags = ["unstable_ts_config"];
+
+            it("should find and load eslint.config.ts when present", async () => {
+
+                const cwd = getFixturePath("ts-config-files", "ts");
+
+                eslint = new ESLint({
+                    cwd,
+                    flags
+                });
+
+                const results = await eslint.lintText("foo");
+
+                assert.strictEqual(eslint.hasFlag("unstable_ts_config"), true);
+                assert.strictEqual(results.length, 1);
+                assert.strictEqual(results[0].messages.length, 1);
+                assert.strictEqual(results[0].messages[0].severity, 2);
+                assert.strictEqual(results[0].messages[0].ruleId, "no-undef");
+
+            });
+
+            it("should load eslint.config.ts when we have \"type\": \"commonjs\" in nearest `package.json`", async () => {
```

Comment:
> If you would like to test that `"type": "commonjs"` in `package.json` is ignored it would be better to have ESM specific syntax in `eslint.config.ts`.

I do that [here](https://github.com/eslint/eslint/blob/f2b07cd738c3790a0dc097aa363ad385ef89c480/tests/lib/eslint/eslint.js#L5219).